polar-core/src/bindings.rs

```diff
@@ -16,9 +16,15 @@ pub struct Binding(pub Symbol, pub Term);
 pub type BindingStack = Vec<Binding>;
 pub type Bindings = HashMap<Symbol, Term>;
 
-pub type Bsp = usize;
+pub type Bsp = Bsps;
 pub type FollowerId = usize;
 
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Bsps {
```

Comment:
Having the indirection through `Bsp` makes it easier to change the type in the future. It made it pretty easy for me to change from `usize` to a struct rather than making updates everywhere.